lib/fswatcher/fswatcher.go

```diff
@@ -125,16 +129,16 @@ func New(id string, cfg *config.Wrapper, ignores *ignore.Matcher) (Service, erro
 func (w *watcher) setupBackend() error {
 	absShouldIgnore := func(absPath string) bool {
 		if !isSubpath(absPath, w.folderPath) {
-			return true
+			panic(fmt.Sprintf("bug: Notify backend is processing a change outside of the root dir of folder %v", w.folderCfg.Description()))
 		}
 		relPath, _ := filepath.Rel(w.folderPath, absPath)
 		return w.folderIgnores.ShouldIgnore(relPath)
 	}
 	if err := notify.WatchWithFilter(filepath.Join(w.folderPath, "..."), w.backendEventChan, absShouldIgnore, w.eventMask()); err != nil {
 		notify.Stop(w.backendEventChan)
 		close(w.backendEventChan)
-		if isWatchesTooFew(err) {
-			err = watchesLimitTooLowError(w.folderDescription)
+		if isOutOfFileHandles(err) {
+			err = watchesLimitTooLowError(w.folderCfg.Description())
```

Comment:
The error wrapper func should also follow the name change. Also it's probably not handles that you run out of, so perhaps should be something more generic isOutOfCapacity or isHittingLimits or something like that (not sure what)